src/hashTable.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const ARBITRARY_ZIP_CODE_HASH = 94607
+const LENGTH_OF_ALPHABET = 26
+const node = ( key, value ) => {
+  let obj = {}
+  obj[ key ] = value
+  return obj
+}
+
+export default class HashTable {
+
+  constructor() {
+    this.length = 0
+    this.collection = {}
+  }
+
+  put( key, value ) {
+
+    if ( !value ) {
+      throw new Error( 'Missing necessary parameter(s)' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key must be a string' )
+    }
+
+    let index = HashTable.hash( key ) % LENGTH_OF_ALPHABET
+    let newNode = node( key, value )
+
+    if ( this.collection[index] ) {
+      this.collection[index].push( newNode )
+    }
+    else {
+      this.collection[index] = [ newNode ]
+    }
+
+    this.length++
+
+  }
+
+  get( key ) {
+
+    if ( !key ) {
+      throw new Error( 'No key given' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key given must be a string' )
+    }
+
+    let index = HashTable.hash( key ) % LENGTH_OF_ALPHABET
+
+    if( this.collection[ index ] ) {
+      let keyFilter = this.collection[ index ].filter( node =>
+        Object.keys( node )[0] === key )
+      if( keyFilter.length > 0 ) {
+        return Object.values(keyFilter[0])[0]
+      }
+      else {
+        return null
+      }
+    }
+    else {
+      return null
+    }
+
+  }
+
+  contains( key ) {
+
+    if ( !key ) {
+      throw new Error( 'No key given' )
+    }
+    if ( typeof key !== 'string' ) {
+      throw new Error( 'Key given must be a string' )
+    }
+
+    return this.get( key ) !== null
+
+  }
+
+  iterate( callback ) {
+
+    if ( typeof callback !== 'function' ) {
+      throw new Error( 'No function given' )
+    }
+
+    Object.keys(this.collection).forEach( key => this.collection[ key ].forEach( node => callback( Object.keys( node )[0], Object.values( node )[0] ) ) )
```

Comment:
this line is too long. Consider this formatting:

```js
Object.keys(this.collection)
  .forEach( key => 
    this.collection[ key ].forEach( node =>
      callback( Object.keys( node )[0], Object.values( node )[0] ) 
  ) 
)
```


Also again you are using this `Object.keys(thing)[0]` pattern again. I think you should change how you're storing your "nodes".

Consider changing this 

```
const node = (key, value) => {
  let obj = {}
  obj[ key ] = value
  return obj
}
```

to this

```
const node = (key, value) => {
  return {key: key, value: value};
}
```

so that later when you have a node you can do this `node.key` instead of `Object.keys(node)[0]` and `node.value` instead of `Object.values(node)[0]`